HW1/P2/P2b.py

```diff
@@ -1,3 +1,19 @@
 from P2 import *
 
-# Your code here
+# initialize spark context
+sc = pyspark.SparkContext("local[4]", "Spark1")
+
+# create RDD of pixel location tuples
+i_vals = sc.parallelize(range(2000))
+ij_vals = i_vals.cartesian(i_vals)
+
+# associate mandelbrot values with each pixel
+ijxy_vals = ij_vals.zip(ij_vals).mapValues(lambda (i,j): (j/500 - 2, i/500 - 2))
+mandel_vals = ijxy_vals.partitionBy(100).mapValues(lambda xy: mandelbrot(*xy))
+#draw_image(mandel_vals)
+#draw_partitions(mandel_vals)
+
+plt.hist(sum_values_for_partitions(mandel_vals).collect(), bins=200, range=(0,2.5e7))
```

Comment:
I'm not sure why you're explicitly setting the range for the histogram, it would be clearer if you let it automatically choose the range, and see the distribution of the iterations per partition much more closely. 

partitionBy(100) will randomize the xy values that each partition performs the computation so I agree with your solution. I did it slightly differently and it seemed to give a slightly better distribution (narrower, more centered around the mean), when I just partitioned by taking a random number between 0 and 99 and assigning xy to that partition.